mock/service/controller.go

```diff
@@ -227,6 +227,11 @@ func (s *service) ControllerPublishVolume(
 	v.VolumeContext[ReadOnlyKey] = roVal
 	s.vols[i] = v
 
+	if volInfo, ok := MockVolumes[req.VolumeId]; ok {
+		volInfo.ISPublished = true
```

Comment:
`ISPublished` is used for NodePublish/Unpublish, not for ControllerPublish. If you need to check for attached volumes before resizing them, add a separate IsControllerPublished flag to `volInfo`.

Which brings me to question, expanding an attached and unmounted volume is online or offline resize?

Reply:
it is still considered online.

Reply:
ack.  the offline check does uses `IsPublished` check, so I will update that code too.